libusb/os/windows_hotplug.c

```diff
@@ -0,0 +1,296 @@
+#include "libusbi.h"
+#include "threads_windows.h"
+
+#include "windows_common.h"
+#include "windows_hotplug.h"
+
+#include <stdio.h>
+#include <dbt.h>
+#include <usbiodef.h>
+
+static HWND windows_event_hWnd;
+static HANDLE windows_event_thread_handle;
+static DWORD WINAPI windows_event_thread_main(LPVOID lpParam);
+static LRESULT CALLBACK windows_proc_callback(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
+
+#define log_error(operation) do { \
```

Comment:
Why is is a macro instead of a function?

Reply:
usbi_err will use caller function to log from which function the error is printed. If I use a function here, all logs would be printed as if originated from this function. By using a macro, I do not modify the call stack, hence I have the correct function context printed for me by usbi_err. 

This is also why I tried to kept my messages as short as possible, adding only what cannot be already deduced from what is included with usbi_err: we already get thread ID and caller function. I am adding Windows API function causing the error + Windows full error message.

I'll add "event thread: " following you request because thread ID of event thread is only printed at Debug level.

Reply:
Is this macro something that can be reused, and moved somewhere else?

I had made more notes here (I made an offline review) but they may be already be addressed by Fabian and your conversation:
These are errors that never should happen? So meant for debugging?
Or is a more explanatory message useful in some cases?

Reply:
@tormodvolden : now that you're mentioning it, I can see several usages in event_windows.c and windows_common.c, e.g.  : `usbi_warn(NULL, "SetWaitableTimer failed: %s", windows_error_str(0));` where similar code is used explicitly, without simplifying macro. We could rename my macro to `usbi_windows_err` or something and use it wherever it makes sens.
As for this PR: I tried to keep changes to existing code down to a bare minimum, refraining myself from touching things along the way. If you think it is a good idea, I could as well make changes for reusing a similar macro everywhere, thought it is not related to "windows hotplug" strictly speaking. Let me know.

As for the error: I don't really know how to qualify them: they should never happen, but if they do it is fatal (hotplug won't work) for most of them, so I guess it worth signaling it to the user with the higher importance log level.